fakebook/src/app/component/newsfeed/newsfeed.component.ts

```diff
@@ -22,23 +22,26 @@ export class NewsfeedComponent implements OnInit {
     ) {}
 
   ngOnInit(): void {
-    this.getUser();
     this.route.params.subscribe((params) => {
-      if(params['id'] !== undefined) {
+      let loadUserFeed = params['id'] !== undefined;
+      this.getPosts(loadUserFeed);
+      if(!loadUserFeed) {
         this.postId = +params['id'];
         if(this.postId !== undefined && !isNaN(this.postId)) {
           this.getPostById();
         }
-      } else {
-        this.getPosts();
-      }
+      } 
     });
   }
 
-  getPosts(): void {
+  getPosts(loadUserFeed : boolean): void {
     this.newsfeedService
-      .getPosts()
-      .subscribe((gotPosts) => (this.posts = gotPosts));
+      .getUser()
+      .subscribe((gotUser) => { 
+        this.user = gotUser;
+         this.newsfeedService.getPosts(gotUser?.followers!)
+         .subscribe((gotPosts) => (this.posts = gotPosts));
```

Comment:
this isn't good, so at least open a github issue about it -
example link on the subject
https://www.thinktecture.com/en/angular/rxjs-antipattern-1-nested-subs/

Reply:
Created issues #90.  I will admit that it doesn't feel nice at all and that it definitely needs a better implementation.